src/views/Nft/market/hooks/useCollectionNfts.tsx

```diff
@@ -0,0 +1,167 @@
+import { useEffect, useState } from 'react'
+import { ApiResponseCollectionTokens, ApiSingleTokenData, NftAttribute, NftToken } from 'state/nftMarket/types'
+import { useGetNftFilters, useGetNftOrdering, useGetNftShowOnlyOnSale } from 'state/nftMarket/hooks'
+import {
+  fetchNftsFiltered,
+  getMarketDataForTokenIds,
+  getNftApi,
+  getNftsFromCollectionApi,
+  getNftsMarketData,
+} from 'state/nftMarket/helpers'
+// eslint-disable-next-line camelcase
+import useSWRInfinite from 'swr/infinite'
+import isEmpty from 'lodash/isEmpty'
+import uniqBy from 'lodash/uniqBy'
+import { REQUEST_SIZE } from '../Collection/config'
+
+interface ItemListingSettings {
+  field: string
+  direction: 'asc' | 'desc'
+  showOnlyNftsOnSale: boolean
+  nftFilters: Record<string, NftAttribute>
+}
+
+const fetchTokenIdsFromFilter = async (address: string, settings: ItemListingSettings) => {
+  const filterObject: Record<string, NftAttribute> = settings.nftFilters
+  const attrParams = Object.values(filterObject).reduce(
+    (accum, attr) => ({
+      ...accum,
+      [attr.traitType]: attr.value,
+    }),
+    {},
+  )
+  const attrFilters = !isEmpty(attrParams) ? await fetchNftsFiltered(address, attrParams) : null
+  return attrFilters ? Object.values(attrFilters.data).map((apiToken) => apiToken.tokenId) : null
+}
+
+const fetchMarketDataNfts = async (
+  address: string,
+  settings: ItemListingSettings,
+  page: number,
+): Promise<NftToken[]> => {
+  const tokenIdsFromFilter = await fetchTokenIdsFromFilter(address, settings)
+  const whereClause = tokenIdsFromFilter
+    ? {
+        collection: address.toLowerCase(),
+        isTradable: true,
+        tokenId_in: tokenIdsFromFilter,
+      }
+    : { collection: address.toLowerCase(), isTradable: true }
```

Comment:
Should reflect to `showOnlyNftsOnSale`

Reply:
fetchMarketDataNfts is called when showOnlyNftsOnSale is true, should we put that condition inside that method?

Reply:
umm.. weird nfts are missing if you select filters

Reply:
Fixed 👍